core-bundle/src/Resources/contao/classes/Frontend.php

```diff
@@ -332,24 +332,32 @@ public static function getRootPageFromUrl()
 			$strError = 'No root page found (host "' . Environment::get('host') . '", languages "' . implode(', ', Environment::get('httpAcceptLanguage')) . '")';
 		}
 
+		$objRequest = Request::create($strUri);
+		$objRequest->headers->set('Accept-Language', $accept_language);
+
 		try
 		{
-			$objRequest = Request::create($strUri);
-			$objRequest->headers->set('Accept-Language', $accept_language);
-
 			$arrParameters = System::getContainer()->get('contao.routing.nested_matcher')->matchRequest($objRequest);
-			$objRootPage = $arrParameters['pageModel'] ?? null;
-
-			if (!$objRootPage instanceof PageModel)
-			{
-				throw new MissingMandatoryParametersException('Every Contao route must have a "pageModel" parameter');
-			}
 		}
 		catch (RoutingExceptionInterface $exception)
 		{
-			$logger->log(LogLevel::ERROR, $strError, array('contao' => new ContaoContext(__METHOD__, 'ERROR')));
+			try
+			{
+				$arrParameters = System::getContainer()->get('contao.routing.nested_404_matcher')->matchRequest($objRequest);
+			}
+			catch (RoutingExceptionInterface $exception)
+			{
+				$logger->log(LogLevel::ERROR, $strError, array('contao' => new ContaoContext(__METHOD__, 'ERROR')));
+
+				throw new NoRootPageFoundException('No root page found', 0, $exception);
+			}
+		}
+
+		$objRootPage = $arrParameters['pageModel'] ?? null;
 
-			throw new NoRootPageFoundException('No root page found', 0, $exception);
+		if (!$objRootPage instanceof PageModel)
```

Comment:
Why not `if (!isset($arrParameters['pageModel'])) {`?

Reply:
I think mostly because that tells my IDE that `$objRootPage` is a `PageModel` later in the code 🤔

Reply:
also, this has been the code before, that didn't change (only rearranged) 😉